### PR TITLE
Refactor and consolidate the dotnet images

### PR DESCRIPTION
**Pull request by mattmoor**
This change has two main parts:
1. It combines `dotnet-{runtime,sdk}` directories into one `dotnet` directory that produces both images,
2. It refactors `dotnet` to follow the style we have been using based on #1172 to consolidate configs and remove version tagging logic.

### CODE REVIEW

File: images/dotnet/configs/runtime/main.tf [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    apko = { source = "chainguard-dev/apko" }
+  }
+}
+
+# TODO: Create a dotnet-runtime virtual that we can track.
+
+variable "extra_packages" {
+  description = "The additional packages to install (e.g. dotnet-7-runtime)."
+  default     = ["dotnet-7-runtime"]
+}
```
Comment (mattmoor) [inline images/dotnet/configs/runtime/main.tf:12]: once https://github.com/wolfi-dev/os/pull/5098 lands we can switch this

File: images/dotnet/configs/sdk/main.tf [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    apko = { source = "chainguard-dev/apko" }
+  }
+}
+
+# TODO: Create a dotnet-sdk virtual that we can track.
+
+variable "extra_packages" {
+  description = "The additional packages to install (e.g. dotnet-7-sdk)."
+  default     = ["dotnet-7-sdk"]
+}
```
Comment (mattmoor) [inline images/dotnet/configs/sdk/main.tf:12]: Once https://github.com/wolfi-dev/os/pull/5098 lands we can switch this

**Comment by mattmoor** [comment]
waiting on: https://github.com/wolfi-dev/os/actions/workflows/build.yaml

**Comment by mattmoor** [comment]
This should be RFAL

**Review by tcnghia — APPROVED** [review]

**Pull request merged**